pkg/storaged/iscsi-panel.jsx

```diff
@@ -30,216 +30,220 @@ import { dialog_open, TextInput, PassInput, SelectRow } from "./dialog.jsx";
-                            ],
-                            Action: {
-                                Title: _("Change"),
-                                action: function (vals) {
-                                    return client.manager_iscsi.call('SetInitiatorName',
-                                                                     [vals.name,
-                                                                         { }
-                                                                     ]);
-                                }
-                            }
-                        });
-                    });
-        }
+        return a_name.localeCompare(b_name);
+    }
 
-        function cmp_session(path_a, path_b) {
-            const a = client.iscsi_sessions[path_a];
-            const b = client.iscsi_sessions[path_b];
-            const a_name = a.data.target_name || "";
-            const b_name = b.data.target_name || "";
+    function make_session(path) {
+        const session = client.iscsi_sessions[path];
 
-            return a_name.localeCompare(b_name);
+        function iscsi_remove() {
+            if (options.disarm)
+                options.disarm();
+            return session.Logout({ 'node.startup': { t: 's', v: "manual" } });
         }
 
-        function make_session(path) {
-            const session = client.iscsi_sessions[path];
+        let actions = null;
+        if (options.armed)
+            actions = (
+                <StorageButton kind="danger" onClick={iscsi_remove}>
+                    <TrashIcon className="ct-TrashIcon" />
+                </StorageButton>
+            );
 
-            function iscsi_remove() {
-                self.setState({ armed: false });
-                return session.Logout({ 'node.startup': { t: 's', v: "manual" } });
-            }
+        return {
+            client,
+            actions,
+            kind: "array",
+            name: session.data.target_name || "",
+            key: path,
+            detail: session.data.persistent_address + ":" + session.data.persistent_port
+        };
+    }
 
-            let actions = null;
-            if (self.state.armed)
-                actions = (
-                    <StorageButton kind="danger" onClick={iscsi_remove}>
-                        <TrashIcon className="ct-TrashIcon" />
-                    </StorageButton>
-                );
+    return Object.keys(client.iscsi_sessions).sort(cmp_session)
+            .map(make_session);
+}
 
-            return {
-                client,
-                actions,
-                kind: "array",
-                name: session.data.target_name || "",
-                key: path,
-                detail: session.data.persistent_address + ":" + session.data.persistent_port
-            };
-        }
+export class IscsiPanel extends React.Component {
+    constructor() {
+        super();
+        this.state = { armed: false };
+    }
+
+    render() {
```

Comment:
This would be an excellent candidate for a functional component.

Reply:
Yep, it's also a excellent candidate for outight removal. :-)